src/resource-scheduling/simulator.md

```diff
@@ -0,0 +1,90 @@
+# Simulator
+
+## Overview
+
+With its flexibility and significant benefits of reducing the time and the
+cost, the simulator plays an important role in studying and designing a
+computer architecture. It is often used to validate specific design schemes
+and evaluate the effectiveness of design schemes.
+
+## Workflow
+
+Since this book is mainly targeted towards the people who are interesting in
+TiKV, we will focus on using the simulator to deal with the scheduling
+problem.
+In general, when there is a lack of resources or the problem is hard to
+reproduce, we might consider using the simulator to make it.
+
+A simulation for the scheduling problem in the distributed system
+usually consists of the following steps:
+
+1. Define the system model of the simulator.
+2. Set up the simulation environment.
+3. Run the simulation.
+4. Inspect the result to check whether it is in line with the expectation.
+
+The first step is mainly to figure out which part of your system you want to
+simulate. And the model should be as simple as possible. In the second step,
+you should set up the environment including the scale of your system and the
+characteristics of the workload. In the third step, the simulation will run
+and provide the scheduling output. In the final step, you can check the
+result and dig into the scheduling problem if the result is not as expected.
+
+## PD simulator
+
+In PD, we also need the simulator to locate a scheduling problem.
+The simulator can be used to simulate a large-scale cluster and scenarios
+with different users.
+And for some special scenarios, we can keep them so that we can quickly
```

Comment:
Hard to understand this sentence :(

Reply:
+1.  What are special scenarios?  Also "keep them" is too general, maybe we need to indicate keep them (where, as what?)

Reply:
How about directly using different scenarios? I am not sure if it is appropriate to list some concrete scenarios here.